webui/react/src/components/NavigationSideBar.tsx

```diff
@@ -107,8 +109,16 @@ export const NavigationItem: React.FC<ItemProps> = ({
 const NavigationSideBar: React.FC = () => {
   // `nodeRef` padding is required for CSSTransition to work with React.StrictMode.
   const nodeRef = useRef(null);
-  const { agents, auth, cluster: overview, resourcePools, info, pinnedWorkspaces } = useStore();
+
+  const { auth, resourcePools, info, pinnedWorkspaces } = useStore();
   const { ui } = useUI();
+  const agents = useAgents();
+  const overview = useClusterOverview();
+  const clusterStatus = Loadable.match(Loadable.all([agents, overview]), {
+    Loaded: ([agents, overview]) => clusterStatusText(overview, resourcePools, agents),
+    NotLoaded: () => undefined, // TODO show spinner when this is loading
```

Comment:
I couldn't find a spinner that actually worked in this spot so I left it a TODO. As a note: this label doesn't show up on latest-master which is annoying for testing.